src/builtins/core/duration.rs

```diff
@@ -264,6 +264,27 @@ impl Duration {
         Self { date, time }
     }
 
+    pub(crate) fn try_new_from_durations(
```

Comment:
note: this will probably be removed when Duration's representation is updated.

Reply:
That's fine. The validity check needs to stick around.